### ISSUE TITLE
Checking account transaction window shows up for a stock account

### ISSUE THREAD

**Issue by kannan-balasubramanian** [moneymanagerex/moneymanagerex#6810]
MMEX version:
 - [ ] 1.7.0
 - [X] 1.8.0
 - [ ] 1.8.1
 - [ ] 1.8.2
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

Traditional transaction window showing up for stock account when a new stock transaction is added.
![Screenshot 2024-07-02 140726](https://github.com/moneymanagerex/moneymanagerex/assets/30168417/188c9c4e-e473-40de-8d1b-25b32794751b)
![Screenshot 2024-07-02 140738](https://github.com/moneymanagerex/moneymanagerex/assets/30168417/2d8e6b38-3dc8-4f23-8fc1-2ba385d93934)

But when the same is done at the investment account level transaction list, then the correct transaction window shows up.
![Screenshot 2024-07-02 140924](https://github.com/moneymanagerex/moneymanagerex/assets/30168417/5d0e572e-50d4-4d53-8616-042f8e648035)
![Screenshot 2024-07-02 140942](https://github.com/moneymanagerex/moneymanagerex/assets/30168417/a2efc8c7-ed9b-40a2-ad17-7234c5c849c1)

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Open the application
2. Expand "Stock Portfolios"
3. Expand the investment account
4. Click on the individual stock/mutual fund account
5. On the right-hand side transaction area, right-click, and select "New Withdrawal..."
6. The normal checking account transaction window shows up instead of stock/mutual fund transaction window.

Expected result:
![Screenshot 2024-07-02 140942](https://github.com/moneymanagerex/moneymanagerex/assets/30168417/a2efc8c7-ed9b-40a2-ad17-7234c5c849c1)

Actual result:
![Screenshot 2024-07-02 140738](https://github.com/moneymanagerex/moneymanagerex/assets/30168417/2d8e6b38-3dc8-4f23-8fc1-2ba385d93934)
# Additional information

If you have a backtrace for a crash or a warning, paste it here.
No crashes experienced.

**Comment by Lanna253** [comment]
I don't have a PC or laptop as for bug 🪲 I don't know I'm not sure it's
difficult to work on a small device it's hard to locate meun button it's
why I'm having issues with getting things done

**Comment by n-stein** [comment]
This is as designed. 

Investment panel = stock transactions
Share account panel = account transactions

The "Share Account" is really just a standard account to record transactions which derive from your shares _without impacting the shares themselves_. For that purpose you can log normal transactions in the account.

For example, dividend income comes from holding shares but doesn't directly impact the number of shares you hold -- i.e. you don't sell shares to receive dividend income. You may choose to reinvest the dividend, but that is a separate transaction which _does_ impact the shares you hold. So to record a dividend you would record a standard deposit transaction in the share account (not a stock transaction). This is the dividend paid. Then if you reinvest the dividend you record a withdrawal stock transaction from the investment screen, or alternatively if you withdraw the dividend you would record a standard transfer transaction from the share account to another account.

Note the same is true for assets. Asset transactions are recorded from the asset panel and normal/standard transactions in the asset account panel.

**Comment by kannan-balasubramanian** [comment]
I understand these are standard accounts, to an extent it makes sense. Does this mean there is no export/import functionality with individual share count on specific transactions?

Initially I was planning to import my transaction data with each SIP purchased on a date, but that is where I discovered this behaviour.

**Comment by n-stein** [comment]
> Does this mean there is no export/import functionality with individual share count on specific transactions?

Yes, exactly. Currently there is no share transaction import. This has been requested in #6085.

I am (slowly) working on an overhaul of the stock transaction system, this is on my to-do list.

If you're good with SQL it is possible to add them directly into the database, but the links are not very straightforward.
